src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -312,66 +313,78 @@ public DownloadInstance datafile(@PathParam("fileId") String fileId, @QueryParam
             downloadInstance.setDataverseRequestService(dvRequestService);
             downloadInstance.setCommand(engineSvc);
         }
+        boolean serviceRequested = false;
+        boolean serviceFound = false;
+
         for (String key : uriInfo.getQueryParameters().keySet()) {
             String value = uriInfo.getQueryParameters().getFirst(key);
-            logger.fine("is download service supported? key="+key+", value="+value);
-
-            if (downloadInstance.checkIfServiceSupportedAndSetConverter(key, value)) {
-                // this automatically sets the conversion parameters in 
-                // the download instance to key and value;
-                // TODO: I should probably set these explicitly instead. 
-                logger.fine("yes!");
-                
-                if (downloadInstance.getConversionParam().equals("subset")) {
-                    String subsetParam = downloadInstance.getConversionParamValue();
-                    String variableIdParams[] = subsetParam.split(",");
-                    if (variableIdParams != null && variableIdParams.length > 0) {
-                        logger.fine(variableIdParams.length + " tokens;");
-                        for (int i = 0; i < variableIdParams.length; i++) {
-                            logger.fine("token: " + variableIdParams[i]);
-                            String token = variableIdParams[i].replaceFirst("^v", "");
-                            Long variableId = null;
-                            try {
-                                variableId = new Long(token);
-                            } catch (NumberFormatException nfe) {
-                                variableId = null;
-                            }
-                            if (variableId != null) {
-                                logger.fine("attempting to look up variable id " + variableId);
-                                if (variableService != null) {
-                                    DataVariable variable = variableService.find(variableId);
-                                    if (variable != null) {
-                                        if (downloadInstance.getExtraArguments() == null) {
-                                            downloadInstance.setExtraArguments(new ArrayList<Object>());
+            logger.fine("is download service supported? key=" + key + ", value=" + value);
+            // The loop goes through all query params (e.g. including key, gbrecs, persistentId, etc. )
+            // So we need to identify when a service is being called and then let checkIfServiceSupportedAndSetConverter see if the required one exists
+            if (key.equals("imageThumb") || key.equals("format") || key.equals("variables")) {
+                serviceRequested = true;
+                //Only need to check if this key is associated with a service
+                if (downloadInstance.checkIfServiceSupportedAndSetConverter(key, value)) {
```

Comment:
because this used to run for any params like :persistentId, gbrecs, key it wasn't possible to just put an exception in an else clause (as was commented out in the previous code.